bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -2,6 +2,134 @@ pub extern crate evmc_sys;
 pub use evmc_sys as ffi;
 
 // TODO: Add helpers for host interface
-// TODO: Add convenient helpers for evmc_result
 // TODO: Add convenient helpers for evmc_execute
 // TODO: Add a derive macro here for creating evmc_create
+
+/// EVMC result structure.
+pub struct EvmcResult(Box<ffi::evmc_result>);
+
+impl EvmcResult {
+    /// Initialize a new result struct given a status code, remaining gas, and optional output
+    /// data.
+    pub fn new(status: ffi::evmc_status_code, gasleft: i64, output: Option<&[u8]>) -> Self {
+        EvmcResult(Box::new(ffi::evmc_result {
+            status_code: status,
+            gas_left: gasleft,
+            output_data: if let Some(buf) = output {
+                buf.clone().as_ptr()
```

Comment:
I think `clone().as_ptr()` means the clone can go out of scope and be deallocated:
https://doc.rust-lang.org/std/primitive.slice.html#method.as_ptr

> Returns a raw pointer to the slice's buffer.
>
> The caller must ensure that the slice outlives the pointer this function returns, or else it will end up pointing to garbage.